### ISSUE TITLE
cluster-agent not listening on port 5005 after upgrading to v1.5.1

### ISSUE THREAD

**Issue by JayH5** [DataDog/datadog-agent#4865]
**Describe what happened:**
After upgrading from cluster-agent v1.4.0 to v1.5.1, the cluster-agent doesn't seem to listen on port 5005 at all. There is also an error in the logs for the cluster-agent leader election:
```
2020-02-10 15:01:19 UTC | CLUSTER | INFO | (pkg/collector/runner/runner.go:327 in work) | Done running check kubernetes_apiserver, next runs will be logged every 500 runs
E0210 15:01:26.794853       1 event.go:296] Could not construct reference to: '&v1.ConfigMap{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"datadog-leader-election", GenerateName:"", Namespace:"datadog", SelfLink:"/api/v1/namespaces/datadog/configmaps/datadog-leader-election", UID:"aa7b4fc4-06bd-11ea-a1f6-0298144c6eb0", ResourceVersion:"27197603", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63709319026, loc:(*time.Location)(0x33d7d20)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string{"control-plane.alpha.kubernetes.io/leader":"{\"holderIdentity\":\"datadog-cluster-agent-58b4cdfb9-skfnf\",\"leaseDurationSeconds\":60,\"acquireTime\":\"2020-02-10T15:01:26Z\",\"renewTime\":\"2020-02-10T15:01:26Z\",\"leaderTransitions\":8}"}, OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Data:map[string]string(nil), BinaryData:map[string][]uint8(nil)}' due to: 'no kind is registered for the type v1.ConfigMap in scheme "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection/leaderelection_engine.go:104"'. Will not report event: 'Normal' 'LeaderElection' 'datadog-cluster-agent-58b4cdfb9-skfnf became leader'
I0210 15:01:26.794922       1 leaderelection.go:245] successfully acquired lease datadog/datadog-leader-election
2020-02-10 15:01:26 UTC | CLUSTER | INFO | (pkg/util/kubernetes/apiserver/leaderelection/leaderelection_engine.go:79 in func1) | New leader "datadog-cluster-agent-58b4cdfb9-skfnf"
2020-02-10 15:01:26 UTC | CLUSTER | INFO | (pkg/util/kubernetes/apiserver/leaderelection/leaderelection_engine.go:86 in func2) | Started leading as "datadog-cluster-agent-58b4cdfb9-skfnf"...
```
Running `datadog-cluster-agent status` command reports that it can't connect to `127.0.0.1:5005`. The container process continues running without exiting. The datadog-agent cannot connect to the cluster-agent.

**Describe what you expected:**
With cluster-agent v1.4.0, the cluster-agent starts and listens on port 5005 without reporting the error.

**Steps to reproduce the issue:**
This is the deployment we use for the cluster-agent:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: datadog-cluster-agent
  namespace: datadog
spec:
  selector:
    matchLabels:
      app: datadog-cluster-agent
  template:
    metadata:
      annotations:
        ad.datadoghq.com/datadog-cluster-agent.check_names: '["prometheus"]'
        ad.datadoghq.com/datadog-cluster-agent.init_configs: '[{}]'
        ad.datadoghq.com/datadog-cluster-agent.instances: '[{"prometheus_url": "http://%%host%%:5000/metrics","namespace":
          "datadog.cluster_agent","metrics": ["go_goroutines","go_memstats_*","process_*","api_requests","datadog_requests","external_metrics",
          "cluster_checks_*"]}]'
      labels:
        app: datadog-cluster-agent
      name: datadog-agent
    spec:
      containers:
      - env:
        - name: DD_KUBE_RESOURCES_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: DD_API_KEY
          valueFrom:
            secretKeyRef:
              key: api-key
              name: datadog-secret
        - name: DD_COLLECT_KUBERNETES_EVENTS
          value: "true"
        - name: DD_LEADER_ELECTION
          value: "true"
        - name: DD_CLUSTER_AGENT_AUTH_TOKEN
          valueFrom:
            secretKeyRef:
              key: token
              name: datadog-auth-token
        - name: DD_EXTERNAL_METRICS_PROVIDER_ENABLED
          value: "false"
        image: datadog/cluster-agent:1.5.1
        imagePullPolicy: IfNotPresent
        name: datadog-cluster-agent
      serviceAccountName: dca
```
We use the recommended RBAC config for the `dca` service account without changes: https://github.com/DataDog/datadog-agent/blob/7.17.0/Dockerfiles/manifests/cluster-agent/rbac/rbac-cluster-agent.yaml

**Additional environment details (Operating System, Cloud provider, etc):**
Amazon EKS: Kubernetes 1.13 on Amazon Linux 2

**Comment by uyushkevi** [comment]
Have the same issue with Amazon EKS: Kubernetes 1.14 and `datadog/cluster-agent:latest` image.

**Comment by vboulineau** [comment]
We're able to reproduce this issue, working on a fix.

**Comment by JayH5** [comment]
This is fixed in cluster-agent v1.5.2
